ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/fetch/fetch.tsx

```diff
@@ -10,13 +10,33 @@
  *
  **/
 
-type Param2 = {
+type Options = {
   headers?: object
   [key: string]: unknown
 }
 
-export default function(url: string, { headers, ...opts }: Param2 = {}) {
-  return window.fetch(url, {
+const fetch = window.fetch
+// @ts-ignore
+window.__global__fetch = fetch
+
+// patch global fetch to warn about usage during development
+if (process.env.NODE_ENV !== 'production') {
+  window.fetch = (...args) => {
+    const error = new Error(
+      [
+        `Using 'window.fetch'.`,
+        'Are you sure you want to do this?',
+        `Most code should use 'react-component/utils/fetch' which provides defaults compatible with the backend.`,
```

Comment:
✏️ 
```suggestion
        `Most code should use 'react-component/utils/fetch' which provides default headers compatible with the backend.`,
```

Reply:
I don't think there's any guarantee that in the future this will be just limited to headers.